src/box/sql/build.c

```diff
@@ -2311,6 +2367,104 @@ fk_constraint_change_defer_mode(struct Parse *parse_context, bool is_deferred)
 			  link)->fk_def->is_deferred = is_deferred;
 }
 
+/**
+ * Drop the tuple or field constraint. If there is a tuple constraint with the
+ * given name, it will be dropped. Otherwise, the constraint with the given name
+ * will be searched for in all fields. If it is found, it will be dropped,
+ * otherwise an error is thrown.
+ */
+static void
+sql_drop_core_constraint(struct Parse *parse)
```

Comment:
I spent some time trying to understand if all those drop/create/alter could be done as a single function which simply updates entire record in `_space` and then remembered that some are stored in other spaces (`_fk_constraint`, `_ck_contraint`). I suggest to rename this function to `sql_drop_tuple_constraint()` and `vdbe_emit_alter_constraint()` -> `vdbe_emit_alter_tuple_constraint()`. Similarly to `vdbe_emit_ck_constraint_...()` and `vdbe_emit_fk_constraint_...()`.

Reply:
The new constraint does not use the `_fk_constraint` and `_fk_constraint` system spaces, they are written directly to the space `opts` or space `format`. I redesigned this function so it just finds the constraint and calls a function to drop it.

Reply:
I know, I used the ck/fk_constraint as examples. My point was that the new name is not really consistent with the existing names. The other constraint types mention their names - fk/ck. But this function uses name 'core' - there are no objects like cores. It rather operates on spaces or could even say on tuples, data. The proposal was to change `sql_drop_core_constraint()` -> `sql_drop_space_constraint()` / `sql_drop_tuple_constraint()` / `sql_drop_data_constraint()` / likewise.

Anyway up to you.

Reply:
Thank you for the explanation. I tried renaming this function to `sql_drop_space_constraint()` or `sql_drop_tuple_constraint()` but I think the name is still a bit misleading. Now I named it `sql_constraint_drop()`. I decided to name it that way because that's how we usually call similar functions. At the same time, the `sql_drop_constraint()` function takes its name from the syntax of the `ALTER TABLE DROP CONSTRAINT` operation.

What do you think of the new name?

Reply:
I liked mode `sql_tuple_constraint_drop()` but can live with the new one too.